schema/snapcraft.yaml

```diff
@@ -123,6 +123,11 @@ properties:
             uniqueItems: true
             items:
               type: string
+          aliases:
+            type: array
+            items:
+              type: string
+              pattern: "^[a-zA-Z0-9][-_.a-zA-Z0-9]*$"
```

Comment:
Alright, I say this is fine here since it's currently what snapd supports-- no need to hold up this PR. The validation tools are probably already written, too. If we change the regex we'll have to do it in a coordinated manner.